Kapu's BetterLimbs port

ElorgRHG opened the pull request:
~~_I am 100% qualified to handle everything in this PR, also a good, small PR for a first time contributor, yes_~~
Will be subject to a LOTTA reviews, as currently its almost a 1:1 port.
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
So basically this PR: https://github.com/BeeStation/BeeStation-Hornet/pull/5611, but on shiptest, guess I'll just copy-paste the original PR's description.

MAINTAINER EDIT:
Closes #612
Closes #571

Refactors limbs to retain their information as owners change. Also, makes most unique limbs their own type.
Improves digitigrade code significantly.

Ports digitigrade jumpsuits from ~~Skyrat~~ Oracle
OracleStation/OracleStation#920
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

Removes bad snowflake code, opens up bodypart modding.

_Also apparently allows Vox, yay_

## Changelog
:cl:Kapu1178, Quantum_m, Ziiro, E-MonaRhg, Marksuckerburg
add: Digitigrade legs now support some jumpsuits (more to be added later)
add: Digitigrade legs now support some hardsuits.
add: Almost every unique bodypart sprite is now a whole limb.
add: Limbgrower limbs are now a sickly green
refactor: Digitigrade legs are now full limbs.
refactor: Limbs retain information as they change owners
refactor: Bodypart code is generally less ass
code: Support for bodytype-based clothing.
fix: Lizard snoots no longer poke through hardsuit helmets
fix: CHANGELING TRANSFORM FUNCTIONS PROPERLY
del: IPC and Super Soldier mutation toxin.
code: Krokodil victims are now a subtype of human
/:cl:

~~CL is also 1:1~~
<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

ElorgRHG committed:
Initial, unchecked, most likely broken, commit

ElorgRHG committed:
Runtime killing, teshari parts and fixes

ElorgRHG committed:
Merge remote-tracking branch 'upstream/master' into better-limbs-port

ElorgRHG committed:
This would be a merge conflict possibly

ElorgRHG committed:
Nothing like porting stuff that you know nothing about

Kapu1178 commented on code/modules/surgery/bodyparts/dismemberment.dm:
```diff
@@ -18,7 +18,13 @@
+
+	if(C.stat <= SOFT_CRIT)//No more screaming while unconsious
+		if(IS_ORGANIC_LIMB(affecting))//Chest is a good indicator for if a carbon is robotic in nature or not.
+			C.emote("scream")
```
```suggestion
			INVOKE_ASYNC(C, /mob.proc/emote, "scream")
```

Kapu1178 commented:
Some general notes for maintainers/the author:
1. Kepori/Teshari and Vox will need their own BODYTYPE_KEPORI and BODYTYPE_VOX flags, as they are not humanoid. Make sure to override their bodyparts' `bodytype` var with `BODYTYPE_KEPORI | BODYTYPE_ORGANIC`

2. I removed Synths & Military Synths on bee because their code was shit and nobody used them. They are almost 100% broken with this.

3. I left... alot... of unused vars in here. I need to clean them up. Also, `#define NUMBER_OF_BODYTYPES` is unused and can be removed.

Dimasw99 commented:
Dat removal of IPC mutation toxin tho. IPC shall be forever doomed now if they somehow got gibbed

zestybastard commented:
> o

Doesn't RnD let you print IPC parts, or only organs?

Ms-Mee commented:
> > o
> 
> Doesn't RnD let you print IPC parts, or only organs?

Organs only, yes.

ElorgRHG committed:
Merge remote-tracking branch 'upstream/master' into better-limbs-port

ElorgRHG committed:
Deconflicted stuff + breaks kepori

ElorgRHG committed:
I'm stupid and fixes kepori

ElorgRHG committed:
Also suggested stuff because I forgot

Commit:
Fixes kepori and digi lizard sprites

Commit:
updates to fit the regex

Commit:
nohus

Commit:
please work thank you

Commit:
adds kepori prosthetics

Kapu1178 committed:
Digishoes refactor, Removes unused vars and synths

Commit:
removes problematic extraneous bone code

Commit:
fixes linter error

MarkSuckerberg commented:
FINALLY PASSES CHECKS:tm:

ElorgRHG committed:
Rachnid limbs + can spawn as IPC and jelly people

ElorgRHG commented:
1. ~~Kepori/Teshari and Vox will need their own BODYTYPE_KEPORI and BODYTYPE_VOX flags, as they are not humanoid. Make sure to override their bodyparts' bodytype var with BODYTYPE_KEPORI | BODYTYPE_ORGANIC~~ Done, and there are no Vox... yet.

2. ~~I removed Synths & Military Synths on bee because their code was shit and nobody used them. They are almost 100% broken with this.~~ Also removed

3. ~~I left... alot... of unused vars in here. I need to clean them up. Also, #define NUMBER_OF_BODYTYPES is unused and can be removed.~~ Cleaned up

ElorgRHG committed:
Lazy copy pasting makes good code, yes

MarkSuckerberg committed:
Merge branch 'master' into better-limbs-port

Kapu1178 committed:
Fixes shitcode & Zombie Infections

Commit:
fixes signal runtime, again

Kapu1178 committed:
Merge branch 'master' into better-limbs-port

Kapu1178 committed:
Runtime fix

tiramisuapimancer commented:
Why does this remove lizard, IPC, and rachnid ability to wear underwear !

Kapu1178 committed:
Fixes ethereal runtimes.

Kapu1178 committed:
I hate felinids.

ElorgRHG committed:
Two commits ported from the original in one commit here so its harder to troubleshoot if something breaks

Kapu1178 committed:
Fixes surgery

Kapu1178 committed:
Fixes oopsie

Kapu1178 committed:
Fixes posibrain surgery

ZephyrTFA commented on code/__DEFINES/mobs.dm:
```diff
@@ -71,6 +71,45 @@
 #define DEVIL_BODYPART "devil"
 
+//Bodypart change blocking flags
+#define BP_BLOCK_CHANGE_SPECIES	(1<<0)
```
```suggestion
#define BP_BLOCK_CHANGE_SPECIES (1<<0)
```

Kapu1178 replied:
I HATE this lint.

Kapu1178 committed:
lllllinters

Kapu1178 committed:
Fixes cargo tech digi jumpsuit

Kapu1178 committed:
as() to as anything

Kapu1178 commented:
YOLO
